tezos-sig-gating-contracts/contracts/examples/gatednftminter_simple_unlimited.mligo

```diff
@@ -0,0 +1,130 @@
+  type fa2_extension = {
+      minter: address;   // MINTER ROLE 
+      next_token_id: nat;
+  }
+  type extended_fa2_storage = fa2_extension NFT.storage
+
+  type mint = {
+    owner    : address;
+    token_info : (string, bytes) map;
+  }
+
+  (* SigGating extension - storage *)
+  type storage = extended_fa2_storage SigGatedExtendable.siggated_storage
+
+  type ret = operation list * storage
+
+  module Errors = struct
+      let asset_already_exist = "AssetAlreadyExist"
+  end
+
+  (* FA2 extension - entrypoints *)
+
+  let apply_mint (mint : mint) (s : storage): ret =
+    // Apply MINT
+    let token_id = s.siggated_extension.extension.next_token_id in
+    let new_extension = { s.siggated_extension.extension with next_token_id=token_id+1n } in
+    let () = Assert.Error.assert (Option.is_none (Big_map.find_opt token_id s.siggated_extension.token_metadata)) Errors.asset_already_exist in
+    let new_token_info: NFT.TZIP12.tokenMetadataData = {token_id=token_id;token_info=mint.token_info;} in
+    let new_token_metadata: NFT.TZIP12.tokenMetadata = Big_map.add token_id new_token_info s.siggated_extension.token_metadata in
+    let ext_fa2_s = { s.siggated_extension with token_metadata=new_token_metadata; extension=new_extension} in
+    let new_fa2_s = NFT.set_balance ext_fa2_s mint.owner token_id in
+    [], { s with siggated_extension=new_fa2_s }
+
+  [@entry]
+  let mint (mint : mint) (s : storage): ret =
+    let () = Assert.assert (Tezos.get_sender () = s.siggated_extension.extension.minter) in
+    apply_mint mint s
+
+  [@view]
+  let nextTokenId(p: unit)(s: storage) : nat = 
+      s.siggated_extension.extension.next_token_id
+
+  (* SigGating extension - entrypoints *)
+
+  [@entry]
+  let setSigner(newSigner: address) (s: storage) : ret =
+    let ops, s = SigGatedExtendable.setSigner newSigner s in
+    (ops, s)
+
+  [@view]
+  let txAuthDataSignerAddress(p: unit)(s: storage) : address = 
+      SigGatedExtendable.getSigner p s 
+
+  type txAuthInput = {
+    userAddress: address;   // user address (used to check nonce)
+    expirationBlock: nat;  // expiration date
+    // functionName: string;   // name of the entrypoint of the calldata (for example "%mint")
+    functionArgs: bytes;   // arguments for the entrypoint of the calldata 
+    signerPublicKey: key;     // public key that signed the payload 
+    signature: signature;   // signature of the payload signed by the given public key
+  }
+  
+  // Example of entrypoint which uses 
+  // - verifyTxAuthData function for signature verification (nonce, expiration) 
+  // - process the calldata itself
+  [@entry]
+  let mint_gated (datainput : txAuthInput) (s : storage): ret =
+    let data : SigGatedExtendable.txAuthData = { 
+      userAddress = datainput.userAddress;
+      expirationBlock = datainput.expirationBlock;
+      functionName = "%mint_gated";
+      functionArgs = datainput.functionArgs;
+      signerPublicKey = datainput.signerPublicKey;
```

Comment:
so in the case of a signerManager, is this the address of the signer of the address of the signerManager?

Reply:
wouldn't it be more simple to store the signer here (either EOA or contract) and not require it in the input?